ember-client/src/main/scala/org/http4s/ember/client/EmberClient.scala

```diff
@@ -17,7 +22,11 @@ final class EmberClient[F[_]: Bracket[?[_], Throwable]] private[client] (
     * The first element represents total connections in the pool, the second
     * is a mapping between the number of connections in the pool for each requestKey.
     */
-  def state: F[(Int, Map[RequestKey, Int])] = pool.state
+  def state: F[(Int, Map[(RequestKey, Option[TLSParameters]), Int])] = pool.state
 
   def run(req: Request[F]): Resource[F, Response[F]] = client.run(req)
 }
+
+object EmberClient {
+  val TLSParameters: Key[TLSParameters] = Key.newKey[IO, TLSParameters].unsafeRunSync
```

Comment:
At risk of being accused of Hungarian Notation, I'd put this in a `Keys` object or name it `Key`.